clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -2903,17 +2903,25 @@ def CIR_InlineAsmOp : CIR_Op<"asm", [RecursiveMemoryEffects]> {
         StrAttr:$asm_string,
         StrAttr:$constraints,
         UnitAttr:$side_effects,
-        AsmFlavor:$asm_flavor);
+        AsmFlavor:$asm_flavor,
+        OptionalAttr<ArrayAttr>:$operand_attrs);
 
   let assemblyFormat = [{
     `(`
     $asm_flavor`,`
     `{` $asm_string $constraints `}`
     `)`
+    (`operand_attrs` `=` $operand_attrs^)?
     (`side_effects` $side_effects^)?
     attr-dict
     operands `:` functional-type(operands, results)
-   }];  
+   }];
+
+   let extraClassDeclaration = [{
+    static StringRef getElementTypeAttrName() {
+      return "elementtype";
```

Comment:
This is a bit odd, seems like you are introducing an attribute name so that it can be added in custom fashion as part of the op's default attr dictionary. This already happens when you add a formal attribute parameter, so why not just add it as part of `arguments`? Tablegen generates setters and you could use those in `CIRAsm.cpp`. I don't understand the `elementtype` name as well, can you elaborate a bit more? We also need some more docs to cover `operand_attrs` and the renamed version of `elementtype`.

Reply:
Yes, it's odd indeed. I was "inspired" by the `InlineAsm` operation from the `llvm` dialect, where this `getElementTypeAttrName` is a key in the dictionary of attributes for operand. But looks like we don't need neither this key nor dictionary here. So I removed this once we don't need it